src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -53,7 +66,363 @@ class DataFrameWriter:
+            >>> _ = session.file.put("tests/resources/testCSVspecialFormat.csv", "@mystage", auto_compress=False)
+            >>> # Define the schema for the data in the CSV files.
+            >>> schema = StructType([StructField("ID", IntegerType()),StructField("USERNAME", StringType()),StructField("FIRSTNAME", StringType()),StructField("LASTNAME", StringType())])
+            >>> # Create a DataFrame that is configured to load data from the CSV files in the stage with a given schema and format.
+            >>> df = session.read.load("@mystage/testCSVspecialFormat.csv",format="csv",schema=schema,format_name="csv_format")
+            >>> # Load the data into the DataFrame and return an array of rows containing the results.
+            >>> df.collect()
+            [Row(ID=0, USERNAME='admin', FIRSTNAME=None, LASTNAME=None), Row(ID=1, USERNAME='test_user', FIRSTNAME='test', LASTNAME='user')]
+        """
+        if not path and self._path:
+            path = self._path
+        if format:
+            self.format(format)
+        if partitionBy:
+            self.partitionBy(partitionBy)
+        for key, value in kwargs.items():
+            self.option(key, value)
+        if mode:
+            self.mode(mode)
+        if not self._file_format_type:
+            raise SnowparkClientExceptionMessages.DF_MUST_PROVIDE_FILETYPE_FOR_WRITING_FILE()
+        return self._write_to_location(
+            path, self._file_format_type, self._save_mode.value, block
+        )
+
+    def partitionBy(self, partition_by: Optional[ColumnOrSqlExpr]) -> "DataFrameWriter":
+        """Specifies an expression used to partition the unloaded table rows into separate files. It can be a :class:`Column`, a column name, or a SQL expression."""
+        self._partition_by = partition_by
+        return self
+
+    def option(self, key: str, value: Any) -> "DataFrameWriter":
+        """Sets the specified option in the DataFrameWriter.
+
+        Use this method to configure any
+        `format-specific options <https://docs.snowflake.com/en/sql-reference/sql/create-file-format.html#format-type-options-formattypeoptions>`_
+        and
+        `copy options <https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions>`_.
+        (Note that although specifying copy options can make error handling more robust during the
+        reading process, it may have an effect on performance.)
+
+        Args:
+            key: Name of the option (e.g. ``compression``, ``skip_header``, etc.).
+            value: Value of the option.
+        """
+        if key.upper() == "PATH":
+            self._path = value
+            return self
+        elif key.upper() == "MODE":
+            return self.mode(value)
+        elif key.upper() in ["FORMAT", "TYPE"]:
+            return self.format(value)
+        elif key.upper() in ["PARTITIONBY", "PARTITION_BY"]:
+            return self.partitionBy(value)
```

Comment:
I would err on the side of keeping one key - `PARTITION_BY` for this unless there is a good reason why we need two.

Reply:
It seems that I can do this in spark:
df.write.option("header",True) \
        .option("partitionBy","ts")\
        .mode("overwrite") \
        .csv("/tmp/zipcodes-state")
so leaving this will handle that case

Reply:
@sfc-gh-aalam let me know if you prefer to remove it

Reply:
then let's keep it.

Reply:
we also might have to update line 177 to use `self.partition_by` or create an alias